daemons/fenced/fenced_history.c

```diff
@@ -330,6 +349,46 @@ stonith_local_history_diff(GHashTable *remote_history,
             }
     }
 
+    if (remote_history) {
+        init_stonith_remote_op_hash_table(&stonith_remote_op_list);
+
+        g_hash_table_iter_init(&iter, remote_history);
+        while (g_hash_table_iter_next(&iter, NULL, (void **)&op)) {
+
+            if (stonith_op_state_pending(op->state) &&
+                pcmk__str_eq(op->originator, stonith_our_uname, pcmk__str_casei)) {
+                crm_warn("Failing pending operation %.8s originated by us but "
+                         "known only from peer history", op->id);
+                op->state = st_failed;
+                op->completed = time(NULL);
+                /* use -EHOSTUNREACH to not introduce a new return-code that might
+                   trigger unexpected results at other places and to prevent
+                   remote_op_done from setting the delegate if not present
+                */
+                stonith_bcast_result_to_peers(op, -EHOSTUNREACH, FALSE);
+            }
+
+            updated = TRUE;
+            g_hash_table_iter_steal(&iter);
+            g_hash_table_replace(stonith_remote_op_list, op->id, op);
+            /* we could trim the history here but if we bail
+             * out after trim we might miss more recent entries
+             * of those that might still be in the list
+             * if we don't bail out trimming once is more
+             * efficient and memory overhead is minimal as
+             * we are just moving pointers from one hash to
+             * another
+             */
+        }
+
+        stonith_fence_history_trim();
```

Comment:
We could make this conditional to updated together with notify. But in case we don't have too many entries the call is cheap as the hash-table holds the size without a need to count or something.